README.md

```diff
@@ -48,298 +48,400 @@ The FWS Payments contract enables ERC20 token payment flows through "rails" - au
-|------------------|-------------------------------|-------------------------------------------------------|
-| Streaming        | Future periodic payments      | Subscriptions, ongoing services                       |
-| Fixed            | Immediate one-time payments   | Onboarding fees, bonuses, penalties, termination fees |
+#### Best Practices for Service Providers
 
-#### Best Practices
+This lockup mechanism places clear responsibilities on the service provider to manage risk:
 
-- **Plan ahead:** Estimate both your ongoing (streaming) and one-time payment needs and fund both buckets accordingly.
-- **Monitor balances:** Before making a one-time payment or increasing the streaming rate / period, check that your account is sufficiently funded.
-- **Use cases:** Use streaming lockup for predictable, recurring payments. Use fixed lockup for immediate payments.
-
-This model gives flexibility: this can guarantee ongoing service payments while also handling special, immediate payments as needed—all with clear, enforced separation and safety for both parties.
+-   **Settle Regularly:** You must settle rails frequently. A rail's `lockupPeriod` is a measure of the risk you are willing to take. If you wait longer than the `lockupPeriod` to settle, you allow a client to build up a payment obligation that may not be fully covered by the lockup guarantee if they become insolvent.
+-   **Monitor Client Solvency:** Use the `getAccountInfoIfSettled` function to check if a client is funded. If their `fundedUntilEpoch` is approaching the current epoch, they are at risk.
+-   **Terminate Proactively:** If a client becomes insolvent or unresponsive, terminate the rail immediately. This is the **only way** to activate the safety hatch and ensure you can claim payment from the funds guaranteed by the streaming lockup.
 
 ## Core Functions
 
 ### Account Management
 
+Functions for managing user accounts, including depositing and withdrawing funds. These functions support both ERC20 tokens and the native network token ($FIL) by using `address(0)` as the token address.
+
 #### `deposit(address token, address to, uint256 amount)`
 
-Deposits tokens into a specified account.
+Deposits tokens into a specified account. This is the standard method for funding an account if not using permits. It intelligently handles fee-on-transfer tokens by calculating the actual amount received by the contract.
 
+- **When to use:** Use this for direct transfers from a wallet or another contract that has already approved the Payments contract to spend tokens.
+- **Native Token (FIL):** To deposit the native network token, use `address(0)` for the `token` parameter and send the corresponding amount in the transaction's `value`.
 - **Parameters**:
-  - `token`: ERC20 token contract address
-  - `to`: Recipient account address
-  - `amount`: Token amount to deposit
+  - `token`: ERC20 token contract address.
```

Comment:
```suggestion
  - `token`: ERC20 token contract address (`address(0)` for FIL).
```

maybe we should do this inline, I missed it above on the first read, can be repeated below for each call taking a `token`